homework/index.js

```diff
@@ -30,17 +32,152 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function addRowToTable(parent, rowElements = []) {
+    const row = createAndAppend('tr', parent);
+    rowElements.forEach(elem => {
+      if (elem.type === 'th') {
+        createAndAppend('th', row, { text: elem.text });
+      } else if (elem.type === 'tableDataWithLink') {
+        const tableDataElement = createAndAppend('td', row);
+        createAndAppend('a', tableDataElement, {
+          text: elem.text,
+          href: elem.href,
+          target: '_blank',
+        });
+      } else if (elem.type === 'td') {
+        createAndAppend('td', row, { text: elem.text });
       }
     });
   }
 
+  function buildRepositoryInfo(repository, containerDiv) {
+    const table = createAndAppend('table', containerDiv, {
+      id: 'repository-table',
+    });
+    const tableBody = createAndAppend('tbody', table);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Repository:' },
+      {
+        type: 'tableDataWithLink',
+        text: repository.name,
+        href: repository.html_url,
+        target: '_blank',
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Description:' },
+      {
+        type: 'td',
+        text: repository.description,
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Forks:' },
+      {
+        type: 'td',
+        text: repository.forks_count,
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Last updated:' },
+      {
+        type: 'td',
+        text: new Date(repository.updated_at).toLocaleDateString(),
+      },
+    ]);
+  }
+
+  function buildRepositoryContributors(contributors, containerDiv) {
+    createAndAppend('h2', containerDiv, { text: 'Contributions' });
+    const contributorsList = createAndAppend('ul', containerDiv, { id: 'contributors-list' });
+    contributors.forEach(contributor => {
+      const listItem = createAndAppend('li', contributorsList, {
+        // class: 'contributor-list-item',
```

Comment:
Remember to take out commented-out code (= dead code) from a pull request. In this case, remove the whole (empty) object.